Backend/microservices/authority/src/main/java/com/ms/authority/service/EmailService.java

```diff
@@ -0,0 +1,74 @@
+package com.ms.authority.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.spring5.SpringTemplateEngine;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.io.File;
+
+
+@Service
+@RequiredArgsConstructor
+public class EmailService {
+
+    private final JavaMailSender emailSender;
+    private final SpringTemplateEngine templateEngine;
+    @Value("${spring.mail.username}")
+    private String from;
+
+    public void sendSimpleMessage(
+            String to,
+            String subject,
+            String text)
+    {
+        SimpleMailMessage message = new SimpleMailMessage();
+        message.setFrom(from);
+        message.setTo(to);
+        message.setSubject(subject);
+        message.setText(text);
+        emailSender.send(message);
+    }
+
+    public void sendMessageWithAttachment(
+            String to,
+            String subject,
+            String text,
+            String pathToAttachment)
+            throws MessagingException {
+        MimeMessage message = emailSender.createMimeMessage();
+        MimeMessageHelper helper = new MimeMessageHelper(message, true);
+        helper.setFrom(from);
+        helper.setTo(to);
+        helper.setSubject(subject);
+        helper.setText(text);
+        FileSystemResource file
+                = new FileSystemResource(new File(pathToAttachment));
+        helper.addAttachment("Invoice", file);
+        emailSender.send(message);
+    }
+
+    @Async
+    public void sendActivationLink(String to, String email, String link) throws IllegalStateException, MessagingException {
```

Comment:
Please change signature to sendActivationLink(String email, String name, String link) 
since parameter email actually contains user` name

Reply:
Ok